libs/ledger-live-common/src/bot/engine.ts

```diff
@@ -723,32 +723,41 @@ function transactionTest<T>({
   const timingThreshold = 30 * 60 * 1000;
   // FIXME: .valueOf to do arithmetic operations on date with typescript
   const dt = Date.now().valueOf() - operation.date.valueOf();
-  invariant(dt > 0, "operation.date must not be in in future");
+  botTest("operation.date must not be in future", () =>
+    expect(dt).toBeGreaterThanOrEqual(0)
+  );
   botTest("operation.date less than 30mn ago", () =>
     expect(dt).toBeLessThan(timingThreshold)
   );
-  invariant(!operation.hasFailed, "operation has failed");
+  botTest("operation must not failed", () => {
+    expect(!operation.hasFailed).toBe(true);
+  });
+
   const { blockAvgTime } = account.currency;
 
   if (blockAvgTime && account.blockHeight) {
     const expected = getOperationConfirmationNumber(operation, account);
     const expectedMax = Math.ceil(timingThreshold / blockAvgTime);
-    invariant(
-      expected <= expectedMax,
-      "There are way too much operation confirmation for a small amount of time. %s < %s",
-      expected,
-      expectedMax
+    botTest("low amount of confirmations", () =>
+      invariant(
```

Comment:
Did you forget to remove invariant like the rest of the diff ?

Reply:
not a problem. the invariant will throw an exception as an `expect`, the important part was to wrap it in the "botTest" context. i kept it an invariant so we can see in the log more details, but essentially it's like `if (!condition) throw new Error(msg`

Reply:
Okay thanks